src/posts/posts.service.ts

```diff
@@ -1,20 +1,37 @@
-import { Injectable } from '@nestjs/common';
+import {
+  Injectable,
+  BadRequestException,
+  ForbiddenException,
+} from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
 import { Repository } from 'typeorm';
 
 import { CreatePostInput } from './dto/create-post.input';
 import { Post } from './entities/post.entity';
+import { UpdatePostInput } from './dto/update-post.input';
 import { User } from './../users/entities/user.entity';
 
 @Injectable()
 export class PostsService {
   constructor(
     @InjectRepository(Post) private postsRepository: Repository<Post>,
   ) {}
+
   create(createPostInput: CreatePostInput, user: User) {
     const userId = user.id;
     const newPost = this.postsRepository.create({ ...createPostInput, userId });
 
     return this.postsRepository.save(newPost);
   }
+
+  async update(userId: number, id: number, updatePostInput: UpdatePostInput) {
+    const post = await this.postsRepository.findOne({ where: { id: id } });
```

Comment:
```suggestion
    const post = await this.postsRepository.findOne(id);
```
Như ri thôi được không em

Reply:
Như z bị lỗi này a `Type 'number' has no properties in common with type 'FindOneOptions<Post>'.`